File: buzu/src/test/java/br/net/buzu/pplspec/model/DomainItemTest.java
package br.net.buzu.pplspec.model;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertNotSame;
import static org.junit.Assert.assertTrue;

import org.junit.Test;

import br.net.buzu.pplspec.model.DomainItem;

/**
 * 
 * @author Douglas Siviotti
 * @since 1.0
 */
public class DomainItemTest {

	@Test
	public void testNull() {
		DomainItem item = DomainItem.parse(null);
		assertEquals(null, item);
	}

	@Test
	public void testBlank() {
		DomainItem item = DomainItem.parse("");
		assertEquals("", item.value());
		assertEquals(null, item.label());
	}

	@Test
	public void testValueLabel() {
		DomainItem item = DomainItem.parse("01=abc");
		assertEquals("01", item.value());
		assertEquals("abc", item.label());
	}

	@Test
	public void testOnlyValue() {
		DomainItem item = DomainItem.parse("345");
		assertEquals("345", item.value());
		assertEquals(null, item.label());
	}

	@Test(expected = NullPointerException.class)
	public void testNullPointerException() {
		new DomainItem(null);
	}

	@Test
	public void testObjectMethods() {
		DomainItem item1 = DomainItem.parse("01=abc");
		DomainItem item2 = DomainItem.parse("01=abc");
		DomainItem item3 = DomainItem.parse("01=xyz");
		DomainItem item4 = DomainItem.parse("02=xyz");
		assertEquals(item1, item2);
		assertEquals(item2, item1);
		assertFalse(item1.equals(null));
		assertFalse(item1.equals("abc"));
		assertTrue(item1.equals(item1));
		assertEquals(item1.hashCode(), item2.hashCode());
		assertEquals(item1.toString(), item2.toString());
		assertFalse(item1.equals(item3));
		assertFalse(item1.equals(item4));
		
		assertEquals(0, item1.compareTo(item2));
		assertEquals(0, item1.compareTo(item3));
		assertEquals(-1, item1.compareTo(item4));
		assertEquals(1, item4.compareTo(item1));
	}

}

File: ppl-spec/src/test/kotlin/br/net/buzu/pplspec/model/DomainItemTest.kt
package br.net.buzu.pplspec.model

import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.Disabled
import org.junit.jupiter.api.Test
/**
 *
 * @author Douglas Siviotti
 * @since 1.0
 */
class DomainItemTest {

    @Test
    fun testBlank() {
        val item = domainItemOf("")
        assertEquals("", item.value())
        assertEquals(null, item.label())
    }

    @Test
    fun testValueLabel() {
        val item = domainItemOf("01=abc")
        assertEquals("01", item.value())
        assertEquals("abc", item.label())
    }

    @Test
    fun testOnlyValue() {
        val item = domainItemOf("345")
        assertEquals("345", item.value())
        assertEquals(null, item.label())
    }

    @Test
    fun testObjectMethods() {
        val item1 = domainItemOf("01=abc")
        val item2 = domainItemOf("01=abc")
        val item3 = domainItemOf("01=xyz")
        val item4 = domainItemOf("02=xyz")
        assertEquals(item1, item2)
        assertEquals(item2, item1)
        assertFalse(item1 == null)
        //assertFalse(item1 == "abc")
        assertTrue(item1 == item1)
        assertEquals(item1.hashCode().toLong(), item2.hashCode().toLong())
        assertEquals(item1.toString(), item2.toString())
        assertFalse(item1 == item3)
        assertFalse(item1 == item4)

        assertEquals(0, item1.compareTo(item2).toLong())
        assertEquals(0, item1.compareTo(item3).toLong())
        assertEquals(-1, item1.compareTo(item4).toLong())
        assertEquals(1, item4.compareTo(item1).toLong())
    }

}
